app/Services/VocabularyServiceProvider.js

```diff
@@ -97,9 +97,62 @@ async function getGroupVocabulary(userId, groupId) {
   return vocabulary;
 }
 
+async function updateVocabulary({ name, translations }, userId, vocabularyCardId) {
+  // delete all translations belonging to vocabulary card
+
+  const oldTranslations = await Translation.findAll({
+    where: {
+      userId,
+      vocabularyCardId,
+    },
+  });
+
+  await Promise.all(
+    oldTranslations.map(async (oldTranslation) => {
+      await oldTranslation.destroy();
+    })
+  );
```

Comment:
I think it's more performant to take the destroy where function, since we don't need the deleted translations.
```js
const deleteCount = await Translation.destroy({
  where: {
    userIds,
    vocabularyCardId,
  },
});
```